File: app/src/main/java/com/hjc/reader/bean/db/HistoryDao.java
package com.hjc.reader.bean.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface HistoryDao {

    @Insert
    void insert(History history);

    @Delete
    void delete(History history);


    @Query("DELETE FROM history")
    void deleteAll();

    @Query("SELECT * FROM history")
    List<History> getAllHistory();

}

File: module-main/src/main/java/com/hjc/learn/main/model/HistoryDao.kt
package com.hjc.learn.main.model

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface HistoryDao {

    @Insert
    fun insert(history: History)

    @Delete
    fun delete(history: History)


    @Query("DELETE FROM history")
    fun deleteAll()

    @Query("SELECT * FROM history")
    fun getAllHistory(): List<History>?

}